File: objective-c/sum-of-multiples/sm/Tests/SumOfMultiples.m
#import "SumOfMultiples.h"

@implementation SumOfMultiples
+ (NSNumber *)toLimit:(NSNumber *)limit inMultiples:(NSArray *)multiples {
    NSInteger sum = 0;
    for (NSInteger n = 1; n < limit.integerValue; ++n) {
        NSUInteger isMultiple = [multiples indexOfObjectPassingTest:^BOOL(NSNumber *factor, NSUInteger _i, BOOL *_stop) {
            return factor.integerValue != 0 && n % factor.integerValue == 0;
        }];
        
        if (isMultiple != NSNotFound) sum += n;
    }

    return @(sum);
}
@end

File: swift/sum-of-multiples/Sources/SumOfMultiples/SumOfMultiples.swift
//Solution goes in Sources

class SumOfMultiples {

    static func toLimit(_ limit: Int, inMultiples: [Int]) -> Int {

        (0..<limit).filter {
            for item in inMultiples.filter { $0 != 0 } {
                if $0 % item == 0 {
                    return true
                }
            }
            return false
        }.reduce(0, +)
    }
}